### PR TITLE
LTD 5031 allow searching on item_description

### PR DESCRIPTION
**Pull request by markj0hnst0n**
### Aim

Allow searching on item_description field

[LTD-5031](https://uktrade.atlassian.net/browse/LTD-5031)

[LTD-5031]: https://uktrade.atlassian.net/browse/LTD-5031?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by seijihg — APPROVED** [review]

File: caseworker/cases/views/denials.py [hunk 1]
```diff
@@ -53,7 +53,13 @@ def get_search_filter(self):
             search_filter.append(f'name:"{party["name"]}"')
             search_filter.append(f'address:"{party["address"]}"')
             filter["country"].add(party["country"]["name"])
-        return (" ".join(search_filter), filter)
+
+        if search_filter:
+            default_search_string = f'{" ".join(search_filter)} denial_cle:"" item_description:""'
```
Comment (kevincarrogan) [inline caseworker/cases/views/denials.py:58]: What does the `default` part of this mean?

File: unit_tests/caseworker/cases/test_denials_view.py [hunk 2]
```diff
@@ -120,6 +121,8 @@ def test_search_denials_party_type_ultimate_and_third_party(
         f"address:{party_users[0]['address']}",
         f"name:{party_users[1]['name']}",
         f"address:{party_users[1]['address']}",
+        "denial_cle:",
```
Comment (kevincarrogan) [inline unit_tests/caseworker/cases/test_denials_view.py:124]: Are there quotes missing here? Shouldn’t it be `denial_cle:””`?

File: caseworker/cases/views/denials.py (hunk 1 above)
Comment (markj0hnst0n) [reply caseworker/cases/views/denials.py:58]: `default` in this case refers to the string that search string that populates the search box initally

File: caseworker/templates/case/denial-for-case.html [hunk 3]
```diff
@@ -105,7 +105,7 @@ <h2 class="govuk-heading-l govuk-!-margin-top-6 govuk-!-margin-bottom-3">
                                 <td class="govuk-table__cell">{{ denial.address|safe }}</td>
                                 <td class="govuk-table__cell">{{ denial.country }}</td>
                                 <td class="govuk-table__cell">{{ denial.denial_cle|safe }}</td>
-                                <td class="govuk-table__cell">{{ denial.item_description }}</td>
+                                <td class="govuk-table__cell">{{ denial.item_description|safe }}</td>
```
[the remark this answers is not in the document]
Comment (markj0hnst0n) [reply caseworker/templates/case/denial-for-case.html:108]: it's sanitized in the serializer for the csv uploader, I believe

File: caseworker/cases/views/denials.py (hunk 1 above)
Comment (depsiatwal) [reply caseworker/cases/views/denials.py:62]: that's the wring place this is where all the inputs are sanitised 
Common place we did when we marked the other as SAFE  
https://github.com/uktrade/lite-api/blob/dev/api/external_data/serializers.py#L146

File: caseworker/templates/case/denial-for-case.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (depsiatwal) [reply caseworker/templates/case/denial-for-case.html:108]: that's the wring place this is where all the inputs are sanitised 
Common place we did when we marked the other as SAFE  
https://github.com/uktrade/lite-api/blob/dev/api/external_data/serializers.py#L146

File: caseworker/cases/views/denials.py (hunk 1 above)
Comment (depsiatwal) [reply caseworker/cases/views/denials.py:62]: I see what your doing here put this aren't natural search in my PR we have instructions and the user can just be expected to type if they want to search by these fields.

Comment (markj0hnst0n) [reply caseworker/cases/views/denials.py:62]: removed the search string variables

File: unit_tests/caseworker/cases/test_denials_view.py (hunk 2 above)
Comment (markj0hnst0n) [reply unit_tests/caseworker/cases/test_denials_view.py]: all this has been removed now to keep the search simpler as per @depsiatwal instructions

**Review by depsiatwal — COMMENTED** [review]
looks good might want to merge in dev and update the query string instructions on the form

**Comment by markj0hnst0n** [comment]
The branch is saying 'up to date with dev'. @depsiatwal the instructions already mention item_description as an option so didn't add any more.

**Review by depsiatwal — APPROVED** [review]
lgtm

**Pull request merged**